### PR TITLE
Fix MuSGD crash on non-matrix params like LayerScale

### PR DESCRIPTION
**Pull request by Y-T-G**
`optimizer=MuSGD` crashes on any model with a trainable parameter that has more than 2 dims but is not a conv filter, for example a LayerScale gain of shape `(C, 1, 1)`. `build_optimizer` routes any parameter with `ndim >= 2` into the Muon group, but `muon_update` only knows how to handle 2D matrices and 4D conv filters (which it reshapes to 2D). A 3D parameter falls through untouched and later fails inside Newton-Schulz orthogonalization, since orthogonalizing a `(C, 1, 1)` tensor is meaningless anyway, it is a per-channel scale, not a weight matrix.

Fix: only route `ndim in {2, 4}` parameters to Muon. Everything else (biases, norm weights, LayerScale-style gains) falls through to the existing SGD branch, same as before Muon was added.

Reproduced on a real model with LayerScale params:
```
AssertionError
  File "ultralytics/optim/muon.py", line 37, in zeropower_via_newtonschulz5
    assert G.ndim in {2, 3}
```
With the fix, `optimizer.step()` completes normally and unaffected parameter groups (2D and 4D) are untouched.

Deleted: nothing removed, one condition narrowed from `ndim >= 2` to `ndim in {2, 4}` to match what `muon_update` actually supports.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary

Refined MuSGD parameter handling to apply Muon updates only to supported linear and convolutional weights, with aligned documentation and a version bump to 8.4.108. ⚙️

### 📊 Key Changes

- Updated MuSGD selection to target only:
  - 2D linear weights.
  - 4D convolutional filters, which are reshaped internally for orthogonalization.
- Kept biases, batch normalization parameters, and other tensor dimensions on standard SGD updates.
- Updated YOLO26 training and optimizer documentation to accurately describe MuSGD behavior.
- Bumped the Ultralytics package version from `8.4.107` to `8.4.108`. 📦

### 🎯 Purpose & Impact

- Improves optimizer correctness by limiting Muon-style updates to compatible parameter types.
- Prevents unsupported parameter shapes from receiving orthogonalized updates, improving training stability and predictability.
- Makes the documented YOLO26 training recipe consistent with the implementation.
- Users training YOLO26 with MuSGD or `optimizer=auto` should receive more reliable optimizer behavior, especially during longer training runs. 🚀

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Y-T-G, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your MuSGD fix, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. This is an automated message, and an engineer will assist with any follow-up needed. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, focused fix. The updated routing matches `muon_update`, which only supports 2D tensors and 4D tensors reshaped as convolutional filters; other parameter shapes remain in the existing SGD parameter groups.

**Comment by UltralyticsAssistant** [comment]
🎉 Merged in [#25447](https://github.com/ultralytics/ultralytics/pull/25447)!

Thank you @Y-T-G, with contributions from @glenn-jocher, for refining MuSGD parameter handling and aligning the YOLO26 optimizer documentation. By applying Muon updates only to supported linear and convolutional weights while keeping other parameters on standard SGD, this work improves correctness, stability, and predictability for YOLO26 training.

> “Great things are done by a series of small things brought together.” — Vincent van Gogh

This PR is a great example of how careful implementation and clear documentation combine to make Ultralytics training more reliable. 🚀
